pkg/plugins/resources/yaml/source.go

```diff
@@ -13,43 +13,61 @@ import (
 // Source return the latest version
 func (y *Yaml) Source(workingDir string) (string, error) {
 	// By default workingDir is set to local directory
-	// Merge File path with current workingDir, unless File is an HTTP URL
-	y.spec.File = joinPathWithWorkingDirectoryPath(y.spec.File, workingDir)
+	var fileContent string
+	var filePath string
 
-	// Test at runtime if a file exist
-	if !y.contentRetriever.FileExists(y.spec.File) {
-		return "", fmt.Errorf("the yaml file %q does not exist", y.spec.File)
+	if len(y.files) > 1 {
+		validationError := fmt.Errorf("Validation error in sources of type 'yaml': the attributes `spec.files` can't contain more than one element for conditions")
+		logrus.Errorf(validationError.Error())
+		return "", validationError
 	}
 
 	if y.spec.Value != "" {
 		logrus.Warnf("Key 'Value' is not used by source YAML")
 	}
 
+	// TODO: warn if the boolean 'KeyOnly' is set?
+
 	if err := y.Read(); err != nil {
 		return "", err
 	}
 
-	data := y.currentContent
+	// loop over the only file
+	for theFilePath := range y.files {
+		fileContent = y.files[theFilePath]
+		filePath = theFilePath
+
+		// Merge File path with current workingDir, unless File is an HTTP URL
+		if len(workingDir) > 0 && workingDir != filePath {
```

Comment:
I'm wondering if this `if len(workingDir) > 0 && workingDir != filePath` tests shouldn't be done directly in joinPathWithWorkingDirectoryPath, that would avoid making these tests in multiple locations (and fixes the places where it's missing?)

Reply:
`len(workingDir) > 0` is already tested in `joinPathWithWorkingDirectoryPath`

but I don't understand why `workingDir != filePath` is needed

Reply:
I've encountered some cases where `workingDir` was set to the same value as `filePath`, and got this as final path without this: `test.yaml/test.yaml`

Reply:
> `len(workingDir) > 0` is already tested in `joinPathWithWorkingDirectoryPath`

This one is on me, botched cherry-picking